src/input/MeshAttributes.cpp

```diff
@@ -0,0 +1,169 @@
+/**
+ * @file
+ *  This file is part of PUMGen
+ *
+ *  For conditions of distribution and use, please see the copyright
+ *  notice in the file 'COPYING' at the root directory of this package
+ *  and the copyright notice at https://github.com/SeisSol/PUMGen
+ *
+ * @copyright 2018 LMU-TUM 
+ * @author Thomas Ulrich
+ */
+
+#include "MeshAttributes.h"
+
+MeshAttributes::MeshAttributes(const char* xmlFilename) {
+    init(xmlFilename);
+}
+
+void MeshAttributes::init(const char* xmlFilename) {
```

Comment:
Is there ar reason we have this init function and not put it in the constructor directly?

Reply:
May this can be avoided. But the reason is because of the scope in this if loop:

https://github.com/SeisSol/PUMGen/blob/xmlMerge/src/input/SimModSuite.h#L121

Reply:
Okay